vignettes/migration_from_the_old_crmPack.Rmd

```diff
@@ -12,8 +12,281 @@ vignette: >
 
 ```{r setup, include=FALSE}
 knitr::opts_chunk$set(echo = TRUE)
+library(tibble)
+library(kableExtra)
+library(checkmate)
 ```
 
+The following Vignette describes the changes that were introduced to the `crmPack`
+package as a result of the package's refactoring. One row in below tables
+represents a single-type, consistent change.
+
+# Class and Methods Name Changes
+
+## Classes
+
+```{r class_change_helpers, echo=FALSE}
+
+# A simple helper function that generates change class record.
+ch_class <- function(name = "", new_name = "", comment = "", add_constructor_info = TRUE) {
+  assert_string(name)
+  assert_string(new_name)
+  assert_string(comment)
+  assert_flag(add_constructor_info)
+
+  was_removed <- ifelse(nchar(new_name) == 0, TRUE, FALSE)
+  if (add_constructor_info && !was_removed) {
+    constructor_info <- "\\+ constructor corresponding update"
+    comment <- ifelse(
+      nchar(comment) == 0,
+      constructor_info,
+      paste0(constructor_info, "; ", comment)
+    )
+  }
+  list(name, new_name, "", "", was_removed, comment)
+}
+
+# A simple helper function that generates change slot record.
+ch_slot <- function(class_name,
+                    name = "",
+                    new_name = "",
+                    comment = "",
+                    add_constructor_info = TRUE) {
+  assert_string(class_name)
+  assert_string(name)
+  assert_string(new_name)
+  assert_string(comment)
+  assert_flag(add_constructor_info)
+
+  was_removed <- ifelse(nchar(new_name) == 0, TRUE, FALSE)
+  if (add_constructor_info) {
+    constructor_info <- "\\+ constructor corresponding update"
+    comment <- ifelse(
+      nchar(comment) == 0,
+      constructor_info,
+      paste0(constructor_info, "; ", comment)
+    )
+  }
+  list(class_name, "", name, new_name, was_removed, comment)
+}
+```
+
+
+```{r class_changes, echo=FALSE}
+
+class_changes <- tribble(
+  ~class_name, ~class_new_name, ~slot_name, ~slot_new_name, ~was_removed, ~comment,
+  !!!ch_class("", "positive_number", "to handle strictly positive valued slots
```

Comment:
hm can we please work without the `ch_class()` etc helpers here? as far as I can see they don't do much. It would be better for maintenance to just have the tibble generated without them.
you can also explain in text paragraph before if there are consistent logics in the table instead of pasting with helper function.